src/util/narrow.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2019 Diffblue Limited. All Rights Reserved.
+
+#ifndef CPROVER_UTIL_NARROW_H
+#define CPROVER_UTIL_NARROW_H
+
+#include "invariant.h"
+#include <type_traits>
+
+/// Alias for static_cast intended to be used for numeric casting
+/// Rationale: Easier to grep than static_cast
+template <typename output_type, typename input_type>
+output_type narrow_cast(input_type value)
+{
+  static_assert(
+    std::is_arithmetic<input_type>::value &&
+      std::is_arithmetic<output_type>::value,
+    "numeric_cast is intended only for numeric conversions");
```

Comment:
The function is called `narrow_cast`, but it does make we wonder whether we really need both `narrow_cast` and `numeric_cast(_v)`?

Reply:
Wasn't numeric cast for `expr -> mp_integer`?

Reply:
`narrow_cast` is intended only to suppress compiler warnings. It isn't meant to handle errors. It works exactly like `static_cast`. But it's easier to grep.